src/pages/resultsView/oncoprint/AddTracks.tsx

```diff
@@ -0,0 +1,297 @@
+                                this.props
+                                    .selectedGenericAssayEntitiesGroupByGenericAssayTypeFromUrl &&
+                                this.props
+                                    .selectedGenericAssayEntitiesGroupByGenericAssayTypeFromUrl[
+                                    type
+                                ]
+                                    ? this.props
+                                          .selectedGenericAssayEntitiesGroupByGenericAssayTypeFromUrl[
+                                          type
+                                      ]
+                                    : []
+                            }
+                            onSelectGenericAssayProfile={
+                                this.props.handlers.onSelectGenericAssayProfile
+                            }
+                            onSubmit={this.onGenericAssaySubmit}
+                        />
+                    </MSKTab>
+                );
+            });
+            return tabs;
+        } else {
+            return null;
+        }
+    }
+
+    @computed get clinicalTabText() {
+        if (
+            this.props.store.clinicalTrackOptions.result &&
+            !_.isEmpty(this.props.store.clinicalTrackOptions.result.clinical)
+        ) {
+            return (
+                <div>
+                    {Tab.CLINICAL}
+                    <span style={{ paddingLeft: 5 }}>
+                        <OncoprintDropdownCount
+                            count={
+                                this.props.store.clinicalTrackOptions.result
+                                    .clinical.length
+                            }
+                        />
+                    </span>
+                </div>
+            );
+        } else {
+            return Tab.CLINICAL;
+        }
+    }
+
+    @computed get heatmapTabText() {
+        if (!_.isEmpty(this.props.store.heatmapMolecularProfiles.result)) {
+            return (
+                <div>
+                    {Tab.HEATMAP}
+                    <span style={{ paddingLeft: 5 }}>
+                        <OncoprintDropdownCount
+                            count={
+                                this.props.store.heatmapMolecularProfiles
+                                    .result!.length
+                            }
+                        />
+                    </span>
+                </div>
+            );
+        } else {
+            return Tab.HEATMAP;
+        }
+    }
+
+    @computed get dropdownWidth() {
+        let width = CLINICAL_WIDTH;
+        if (this.props.heatmapMenu) {
+            width += HEATMAP_WIDTH;
+        }
+        const genericAssayTextCount = this.showGenericAssayTabs
+            ? this.genericAssayTabs!.map(tab =>
+                  deriveDisplayTextFromGenericAssayType(tab.key as string)
+              ).join().length
+            : 0;
+        const genericAssayTabsWidth =
```

Comment:
It's difficult to get reference of the whole tabs width, use the estimate width calculated by count of characters.

Reply:
we have a method that measures this, see `getTextWidth` in `packages/cbioportal-frontend-commons/src/lib/TextTruncationUtils.ts`

Reply:
seems fine to me

Reply:
listen to @adamabeshouse

Reply:
sure, will do